商品情報編集機能

shb1204st opened the pull request:
#What
商品情報編集機能の実装

#Why
実装終了いたしましたので、ご確認をお願い致します。

確認のための動画URL
1、ログイン状態の出品者は、商品情報編集ページに遷移できる動画
https://gyazo.com/8c65a57efb7d5de40758d5f0f394046e

2、必要な情報を適切に入力して「変更する」ボタンを押すと、商品の情報を編集できる動画
https://gyazo.com/2450a523d5ebea011265f2b587c4d626

3、入力に問題がある状態で「変更する」ボタンが押された場合、情報は保存されず、編集ページに戻りエラーメッセージが
　  表示される動画
https://gyazo.com/8d1bd7cd5a654bce6ed7d778a041e35c

4、何も編集せずに「変更する」ボタンを押しても、画像無しの商品にならない動画
①元の情報
https://gyazo.com/829512c339859804649520fdb03a796c
②何も編集せずに「変更する」ボタンを押す
https://gyazo.com/2a33586f3826e430b3f0d458cc47702a

5、ログイン状態の場合でも、URLを直接入力して自身が出品していない商品の商品情報編集ページへ遷移しようとすると、
　  商品の販売状況に関わらずトップページに遷移する動画
①他ユーザー情報と商品出品情報
https://gyazo.com/302ae63497e17c6898343786460fb21c
②URLを直接入力して自身が出品していない商品の商品情報編集ページへ遷移しようとすると、トップページに遷移する動画
https://gyazo.com/200bfc13286074af239d0d3a1c6f1913

6、ログアウト状態の場合は、URLを直接入力して商品情報編集ページへ遷移しようとすると、商品の販売状況に関わらず
　  ログインページに遷移する動画
https://gyazo.com/9b2f9ec6ca5956560b33e3b246a6e4f5

7、商品名やカテゴリーの情報など、すでに登録されている商品情報は商品情報編集画面を開いた時点で
　  表示される動画（商品画像・販売手数料・販売利益に関しては、表示されない状態で良い）
①元の情報
https://gyazo.com/829512c339859804649520fdb03a796c
②商品名やカテゴリーの情報など、すでに登録されている商品情報は商品情報編集画面を開いた時点で表示される動画
https://gyazo.com/3541f93feb24b8538e294d4e4d37e0c5

ご確認、よろしくお願い致します。

shb1204st committed:
商品編集機能へのパス設定

shb1204st committed:
商品編集機能のルーティング設定

shb1204st committed:
商品編集機能のコントローター設定

shb1204st committed:
商品編集機能のビュー設定

shb1204st committed:
商品編集機能　コントローラー設定追加

商品出品者以外のログインユーザーが商品の編集（URLにて）をしようとした場合に、トップページに遷移する

shb1204st committed:
Rubocopによる訂正

kotonekurita commented on app/controllers/products_controller.rb:
```diff
@@ -1,6 +1,7 @@
   before_action :authenticate_user!, except: [:index, :show]
-  before_action :set_product, only: [:show]
+  before_action :set_product, only: [:show, :edit, :update]
+  before_action :move_to_index, only: [:edit]
 
```
こちらはupdateアクションにも適用させましょう！

#### 理由：
セキュリティ面を強化するためです。
create、update、destroyアクションは、今回のFURIMAにおいて対応するビューファイルがなく（create.html.erbなど）、一見不正なアクセスができないように思える処理ですが、「検証ツールを使用する」「URLの直接入力をする」といった行為で不正なアクセスができてしまうことがあるためです。

kotonekurita commented on app/views/products/edit.html.erb:
```diff
@@ -7,10 +7,10 @@ app/assets/stylesheets/items/new.css %>
+    <%= form_with model: @product, url:product_path, local: true do |f| %>
 
     <%# インスタンスを渡して、エラー発生時にメッセージが表示されるようにしましょう。%>
-    <%# render 'shared/error_messages', model: f.object %>
+    <%= render 'shared/error_messages', model: f.object %>
     <%# //インスタンスを渡して、エラー発生時にメッセージが表示されるようにしましょう。%>
```
こちらのもともとあるコメントアウトですが、要件の注意事項にありますように実装が完了し不要であれば削除しましょう。
実装完了時に、不要なコメントアウトが存在すると可読性が下がるためです。
もし必要であれば、第三者でもわかるように「残す目的」「なぜ必要か」という理由などを、コメントとして書いておくことをおすすめします。
その他同様のコメントアウトについても削除をお願いいたします。

kotonekurita reviewed (commented):
レビュー依頼ありがとうございます！
#### 1件コメントいたしましたので、ご確認をお願いいたします。

修正後、再度レビュー依頼フォームのご提出をお願いいたします。

shb1204st committed:
pull request5 コントローラーの訂正

before_action :move_to_index, only: [:edit, :update]
updateを追加・・・セキュリティ強化のためで検証ツールを使用して書き換えることも可能になってくるため

shb1204st committed:
pull request5 実装コメントの削除

shb1204st commented:
#What
商品情報編集機能の実装

#Why
pull requesでアドバイスいただいた箇所を修正しました。ご確認をお願い致します。

よろしくお願い致します。

ghost commented:
レビュー依頼ありがとうございます！

いただいたレビュー依頼ですが、修正点の差分が確認できない状況です。
Github Desktopにてコミットとプッシュのし忘れがないかのご確認をお願いいたします。

リモートのGithubにてコミットプッシュの反映にタイムラグが出る場合があり、そちらの影響で反映されていないだけでしたら申し訳ございません。

コミットとプッシュの確認・修正後、再度レビュー依頼のご提出をお願いします。
お手数をおかけしますが、何卒よろしくお願いいたします。

shb1204st commented:
#What
商品情報編集機能の実装

#Why
pull requesでアドバイスいただいた箇所を修正し、コミットプッシュ致しました。ご確認をお願い致します。

よろしくお願い致します。

Hide-Morishita commented:
レビュー依頼ありがとうございます。
LGTMとさせていただきます。実装お疲れ様でした。

マージ後、ローカルのmasterブランチに忘れずにプルしましょう。
その後、新しいブランチを作成し次の実装に取り組んでいただけますと幸いです。
引き続き実装頑張っていきましょう。
